packages/sitecore-jss/src/personalization/personalization-decisions-service.ts

```diff
@@ -0,0 +1,175 @@
+import { AxiosDataFetcher } from './../axios-fetcher';
+import { fetchData, HttpDataFetcher } from './../data-fetcher';
+import debug from '../debug';
+import { extractQueryStringParams } from '../utils/extract-querystring-params';
+
+export interface RenderingPersonalizationDecision {
+  /**
+   * The variant key
+   */
+  variantKey?: string;
+  /**
+   * The error message
+   */
+  errorMessage?: string;
+}
+
+export interface DecisionsContext {
+  /**
+   * The route path
+   */
+  routePath: string;
+  /**
+   * The language
+   */
+  language: string;
+  /**
+   * The rendering identifiers
+   */
+  renderingIds: string[];
+}
+
+export interface PersonalizationDecisionData {
+  /**
+   * The renderings
+   */
+  renderings: {
+    [renderingId: string]: RenderingPersonalizationDecision;
+  };
+}
+
+export interface PersonalizationDecisionsService {
+  /**
+   * Gets personalization decisions
+   */
+  getPersonalizationDecisions(context: DecisionsContext): Promise<PersonalizationDecisionData>;
+}
+
+export type DataFetcherResolver = <T>(config: {
+  /**
+   * The request timeout in milliseconds
+   */
+  timeout?: number;
+}) => HttpDataFetcher<T>;
+
+export type RestPersonalizationDecisionsServiceConfig = {
+  /**
+   * Hostname of decisions service; e.g. http://my.site.core; Default: '', same host as a page
+   */
+  host?: string;
+  /**
+   * Relative path from host to decisions service. Default: /sitecore/api/layout/personalization/decision
+   */
+  route?: string;
+  /**
+   * This value overrides the default service URL.
+   * Note: `host` and `route` options are ignored if `serviceUrl` is set.
+   */
+  serviceUrl?: string;
+  /**
+   * The Sitecore SSC API key your app uses
+   */
+  apiKey: string;
+  /**
+   * The JSS application name
+   */
+  siteName: string;
+  /**
+   * Query string parameters of the current page to pass with request to decision service
+   */
+  currentPageParamsToExtract?: string[];
+  /**
+   * Enables/disables analytics tracking for the Layout Service invocation (default is true).
+   * More than likely, this would be set to false for SSG/hybrid implementations, and the
+   * JSS tracker would instead be used on the client-side: {@link https://jss.sitecore.com/docs/fundamentals/services/tracking}
+   * @default true
+   */
+  tracking?: boolean;
+  /**
+   * The request timeout in milliseconds
+   */
+  timeout?: number;
+  /**
+   * Data fetcher resolver in order to provide custom data fetcher
+   * @see DataFetcherResolver
+   * @see HttpDataFetcher<T>
+   * @see AxiosDataFetcher used by default
+   */
+  dataFetcherResolver?: DataFetcherResolver;
```

Comment:
Does this service need a data fetcher resolver? This pattern was used in layout service because it was fetching data that could either be page data or placeholder data. But looks like this service is always fetching the same type of data. So could be simplified. Then dataFetcher doesn't need to be generic either.